### PR TITLE
Snyder - LB163 bio yield r glu irr

### PR DESCRIPTION
**Pull request by abigailsnyder**
@pkyle do I have the right Wullschleger citation for the  MAX_BIO_YIELD_THA in constants.R?

#470

### CODE REVIEW

**Comment by pkyle** [comment]
Yes that citation is correct.

File: R/constants.R [hunk 1]
```diff
@@ -83,6 +83,12 @@ MAX_HA_TO_CROPLAND <- 3
 # Yield multiplier that goes from the observed yield to the "high" and "low" yields: observed plus or minus observed times this number
 MGMT_YIELD_ADJ <- 0.1
 
+# Maximum biomass yield
+# Source: Wullschleger doi:10.2134/agronj2010.0087 ??
+MAX_BIO_YIELD_THA <- 20
+
+BIO_GJT <- 17.5
```
Comment (bpbond) [inline R/constants.R]: Biogidget? This could use an explanatory comment.

File: R/zchunk_LB163.bio_Yield_R_GLU_irr.R [hunk 2]
```diff
@@ -39,51 +41,160 @@ module_aglu_LB163.bio_Yield_R_GLU_irr_DISABLED <- function(command, ...) {
...
+      filter(iso == "usa") %>%
+      select(GCAM_region_ID) ->
+      USAreg
+
+    # Calculate the base yield, a scaler value:
+    L163.YieldIndex_R_GLU_irr %>%
+      filter(GCAM_region_ID == USAreg$GCAM_region_ID) %>%
+      summarise(maxYieldIndex = max(YieldIndex)) %>%
+      mutate(BaseYieldIndex = MAX_BIO_YIELD_THA / maxYieldIndex) %>%
+      mutate(BaseYieldIndex_GJm2 = BaseYieldIndex * BIO_GJT / CONV_HA_M2) %>%
```
Comment (bpbond) [inline R/zchunk_LB163.bio_Yield_R_GLU_irr.R]: Could combined these into a single `mutate` call

File: R/constants.R (hunk 1 above)
Comment (bpbond) [inline R/constants.R]: FYI in my about-to-opened PR I have 
```R
aglu.MAX_BIO_YIELD_THA <- 20 # Maximum bioenergy (switchgrass) yield allowable, in tons per hectare

aglu.BIO_GJT <- 17.5 # Energy content of biomass
```

Comment (abigailsnyder) [inline R/constants.R]: ah - would you like me to update my code to reflect your naming convention? 
Also, thank you for the comment on BIO_GjT. I had no idea what it was.

Comment (kvcalvin) [inline R/constants.R]: My open PR also has 
```
# Energy content of biomass
aglu.BIO_ENERGY_CONTENT_GJT <- 17.5
```

Comment (bpbond) [inline R/constants.R]: Funny. Ok it's survival of the fittest! Whichever PR gets merged first, the others have to conform :)

Comment (abigailsnyder) [inline R/constants.R]: pretty sure I'll update mine to be more in line with one of yours either way. The aglu tag and additional comments are important.

Comment (bpbond) [inline R/constants.R]: I still kind of like biogidget though.

Comment (abigailsnyder) [inline R/constants.R]: bio_flying_nun is up next

**Comment by bpbond** [comment]
@abigailsnyder ready to merge this whenever you get a chance to address minor comments above

File: R/zchunk_LB163.bio_Yield_R_GLU_irr.R [hunk 3]
```diff
@@ -39,51 +41,160 @@ module_aglu_LB163.bio_Yield_R_GLU_irr_DISABLED <- function(command, ...) {
...
-    #   ... new code with a fix
-    # }
+
+    # Perform computations
+    # old comment: This method follows the same method as LB113, with the exception that
+    #              the yield indices are computed separately for rainfed/irrigated, but
+    #              again against the global average for each crop, across both irrigated
+    #              and rainfed. This method should roughly preserve the global average
+    #              bioenergy yields; what we want to avoid here is increasing the global
+    #              average yields just by separating irr/rfd.
```
Comment (kvcalvin) [inline R/zchunk_LB163.bio_Yield_R_GLU_irr.R]: I know I approved this, but after a conversation with Marshall I want to make a small edit to this comment. Please remove the sentence that starts "This method". This isn't guaranteed to be true as it depends on the ghost share and it isn't necessary.

**Comment by abigailsnyder** [comment]
@bpbond @kvcalvin changes done. Passed check package on my local machine, just waiting on travis

**Comment by bpbond** [comment]
This is just waiting on coverage. Merging.

**Pull request merged**